server/internal/usecase/interactor/asset.go

```diff
@@ -241,8 +260,14 @@ func (i *Asset) Delete(ctx context.Context, aid id.AssetID, operator *usecase.Op
 				return aid, err
 			}
 
-			eOperator := event.OperatorFromUser(operator.User) //TODO: change operator after integration API is implemented
-			if _, err := i.eventFunc(ctx, prj.Workspace(), event.AssetDelete, asset, eOperator); err != nil {
+			var eOp event.Operator
+			if operator.User != nil {
+				eOp = event.OperatorFromUser(*operator.User)
+			}
+			if operator.Integration != nil {
+				eOp = event.OperatorFromIntegration(*operator.Integration)
+			}
+			if _, err := i.eventFunc(ctx, prj.Workspace(), event.AssetDelete, asset, eOp); err != nil {
```

Comment:
For now, implentation of conversion of operators is duplicated. Define a func such `toEventOperator`.

```suggestion
			if _, err := i.eventFunc(ctx, prj.Workspace(), event.AssetDelete, asset, toEventOperator(o)); err != nil {
```